src/smrt/shift_driver.cpp

```diff
@@ -59,27 +59,15 @@ ShiftDriver::ShiftDriver(SP_Assembly_Model assembly,
   add_power_tally(plist, z_edges);
 }
 
-//---------------------------------------------------------------------------//
-// Solve
-//---------------------------------------------------------------------------//
-void ShiftDriver::solve(const std::vector<double>& th_temperature,
-                        const std::vector<double>& coolant_density,
-                        std::vector<double>& power)
+std::vector<double> ShiftDriver::heat_source(double power) const
 {
-  update_temperature(th_temperature);
-
-  // currently does nothing
-  update_density(coolant_density);
-
-  // Rebuild problem (loading any new data needed and run transport
-  d_driver->rebuild();
-  d_driver->run();
-
   // Extract fission rate from Shift tally
   auto sequence = d_driver->sequence();
   auto shift_seq = std::dynamic_pointer_cast<omnibus::Sequence_Shift>(sequence);
   const auto& tallies = shift_seq->tallies();
 
+  std::vector<double> d_power_by_cell_ID;
```

Comment:
Now the name is misleading because it's no longer a data member.
```suggestion
  std::vector<double> power_by_cell_ID;
```